src/osd/PG.cc

```diff
@@ -4619,10 +4650,19 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
         assert(last_update_applied >= scrubber.subset_last_update);
 
         // build my own scrub map
-        ret = build_scrub_map_chunk(scrubber.primary_scrubmap,
-                                    scrubber.start, scrubber.end,
-                                    scrubber.deep, scrubber.seed,
-				    handle);
+	{
+	  epoch_t epoch = get_osdmap()->get_epoch();
+	  unlock();
+	  ret = build_scrub_map_chunk(scrubber.primary_scrubmap,
+				      scrubber.start, scrubber.end,
+				      scrubber.deep, scrubber.seed,
+				      &scrubber.preempted,
+				      handle);
+	  lock();
+	  if (pg_has_reset_since(epoch)) {
+	    return;
```

Comment:
Does it need do cleanup work(call scrub_clear_state(); scrub_unreserve_replicas();)?

Reply:
that will have happend when the pg state changes (see on_change()) already.  if we do something here we might end up clobbering even newer state.